src/components/OnboardingWelcomeVideo.tsx

```diff
@@ -0,0 +1,161 @@
+        videoWidth: number;
+        videoHeight: number;
+    };
+};
+
+type VideoPlaybackStatusEventType = {
+    isLoaded: boolean;
+};
+
+type VideoStatus = 'video' | 'animation';
+
+function OnboardingWelcomeVideo() {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const containerDimensions = useRef<LayoutRectangle>({width: 0, height: 0, x: 0, y: 0});
+    const [isModalVisible, setIsModalVisible] = useState(true);
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const {shouldUseNarrowLayout} = useOnboardingLayout();
+    const [welcomeVideoStatus, setWelcomeVideoStatus] = useState<VideoStatus>('video');
+    const [isWelcomeVideoStatusLocked, setIsWelcomeVideoStatusLocked] = useState(false);
+    const [isVideoLoaded, setIsVideoLoaded] = useState(false);
+    const [videoAspectRatio, setVideoAspectRatio] = useState(VIDEO_ASPECT_RATIO);
+    const {isOffline} = useNetwork();
+
+    useEffect(() => {
+        if (isWelcomeVideoStatusLocked) {
+            return;
+        }
+
+        if (isOffline) {
+            setWelcomeVideoStatus('animation');
+            setIsWelcomeVideoStatusLocked(true);
+        } else if (!isOffline && isVideoLoaded) {
+            setWelcomeVideoStatus('video');
+            setIsWelcomeVideoStatusLocked(true);
+        }
+    }, [isOffline, isVideoLoaded, isWelcomeVideoStatusLocked]);
+
+    const storeContainerDimensions = (event: LayoutChangeEvent) => {
+        containerDimensions.current = event.nativeEvent.layout;
+    };
+
+    const closeModal = useCallback(() => {
+        setIsModalVisible(false);
+        Navigation.goBack();
+    }, []);
+
+    const setAspectRatio = (e: VideoReadyForDisplayEvent | VideoLoadedEventType | undefined) => {
+        if (!e) {
+            return;
+        }
+
+        // TODO: Figure out why on mobile there's e.naturalSize and on web it's e.srcElement
+        if ('naturalSize' in e) {
+            setVideoAspectRatio(e.naturalSize.width / e.naturalSize.height);
+        } else {
+            setVideoAspectRatio(e.srcElement.videoWidth / e.srcElement.videoHeight);
+        }
+    };
+
+    const setVideoStatus = (e: VideoPlaybackStatusEventType) => {
+        setIsVideoLoaded(e.isLoaded);
+    };
+
+    const getWelcomeVideo = () => {
+        if (welcomeVideoStatus === 'video') {
+            const videoWidth = containerDimensions.current.width - 2 * MODAL_PADDING;
+
+            return (
+                <View
+                    style={[
+                        // Prevent layout jumps by reserving height for the video
+                        {height: VIDEO_HEIGHT - 2 * MODAL_PADDING},
+                    ]}
+                >
+                    <VideoPlayer
+                        // Temporary file supplied for testing purposes, to
+                        // be changed when correct one gets uploaded on backend
+                        url="https://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny.mp4"
```

Comment:
The video is ready, you can now use `${CLOUDFRONT_URL}/videos/intro-1280.mp4` (i.e. https://d2k5nsl2zxldvw.cloudfront.net/videos/intro-1280.mp4)